app/assets/javascripts/distributions_and_transfers.js

```diff
@@ -94,11 +94,39 @@ $(function() {
     }
   );
 
-  $(function() {
+  $(function () {
     if (storage_location_required && !control.val()) {
       $("#__add_line_item").addClass("disabled");
     }
 
     request_storage_location_and_populate_item(default_item);
   });
+
+  $('#new_distribution').submit(function (e) {
```

Comment:
@armahillo should we maybe tie this directly to the `onclick` event for the distribution? We keep hitting errors with these js globals that target specific classes or IDs.

Reply:
That's not a bad idea.

I think if we made the main template use a universally computed class / ID (based on controller / action), and added a test to verify that was happening (so when we inevitably re-template again, , we could use CSS selectors for form components in JS that were rooted on that. (so instead of `form#new_distribution`, it would be `body#distributions.new #content .pane form` or something similar)